### PR TITLE
WIP: Calculate the correct Content-Length, if gzip is enabled (fixes #153)

### LINKED ISSUE #153: Header Content-Length with gzip in output method
Hi,

thanks for contributing this great lib.

We using gzip with ob_gzhandler in our framework and use the output function of parsecsv to show or download a generated csv file, but the browser shows an error.

The error is caused by setting the Content-Length for the non gzipped content. If I remove the Content-Length line (448), everything is working fine. 

I see three possible solutions:
1. stop setting the Content-Length
2. add an argument to prevent the output method from setting the Content-Length
3. calculate the correct Content-Length, if gzip is enabled

### PR DESCRIPTION
**Pull request by susgo**
WIP because test is missing. How to test it?

### CODE REVIEW

**Comment by Fonata** [comment]
This comment refers to the 2nd of the 3 currently commits in this PR: Please use `git rebase` instead of `git pull` to keep the history simple. 

Thanks :blush:

File: src/Csv.php [hunk 1]
```diff
@@ -444,14 +444,28 @@ public function output($filename = null, $data = array(), $fields = array(), $de
             $mime = $delimiter === "\t" ?
                 'text/tab-separated-values' :
                 'application/csv';
+
+            // Buffer output:
+            ob_start();
```
Comment (Fonata) [inline src/Csv.php]: I suppose we shouldn't nest `ob_start` calls. Also, they need to come in pairs with `ob_end_*` calls. We need output buffering only if gzip is supported.

File: src/Csv.php [hunk 2]
```diff
@@ -444,14 +444,28 @@ public function output($filename = null, $data = array(), $fields = array(), $de
...
+                && stripos($_SERVER['HTTP_ACCEPT_ENCODING'], "gzip") !== false) {
+                ob_start("ob_gzhandler");
+                echo $flat_string;
+                ob_end_flush();
+            } else {
+                echo $flat_string;
+            }
+
+            // Write the content length
+            header('Content-Length: '.ob_get_length());
```
Comment (Fonata) [inline src/Csv.php]: Wrap `.` operator in spaces.

File: src/Csv.php [hunk 3]
```diff
@@ -444,14 +444,28 @@ public function output($filename = null, $data = array(), $fields = array(), $de
...
             header('Cache-Control: no-cache, must-revalidate');
             header('Pragma: no-cache');
             header('Expires: 0');
             header('Content-Disposition: attachment; filename="' . $filename . '"; modification-date="' . date('r') . '";');
 
-            echo $flat_string;
+            // gzip enabled?
+            if (isset($_SERVER['HTTP_ACCEPT_ENCODING']) 
+                && stripos($_SERVER['HTTP_ACCEPT_ENCODING'], "gzip") !== false) {
+                ob_start("ob_gzhandler");
```
Comment (Fonata) [inline src/Csv.php]: 1. We also need to check if the `zlib` extension is enabled, e.g. by calling `function_exists('ob_gzhandler')`.
2. We need to tell the browser that we compressed: ``header("Content-Encoding: gzip");``

**Comment by susgo** [comment]
You're right. We shouldn't fix it here.

**Pull request closed without merging**